pkg/ddl/index.go

```diff
@@ -1594,31 +1595,12 @@ func (w *addIndexTxnWorker) checkHandleExists(idxInfo *model.IndexInfo, key kv.K
 }
 
 func genKeyExistsErr(key, value []byte, idxInfo *model.IndexInfo, tblInfo *model.TableInfo) error {
-	idxColLen := len(idxInfo.Columns)
+	valueStr, err := local.GenIndexValueFromIndex(key, value, tblInfo, idxInfo)
```

Comment:
Maybe we can move `GenIndexValueFromIndex` into another package? It is not the responsibility of `local` package to build the message of duplicate entry. (In fact the old logic is copied from `ddl` package)